app/src/main/java/org/mozilla/fenix/tabstray/TabsTrayInteractor.kt

```diff
@@ -35,6 +35,15 @@ interface TabsTrayInteractor {
      */
     fun onInactiveDebugClicked(tabs: Collection<TabSessionState>)
 
+    /**
+     * Invoked when [tabId] should be moved to before/after [targetId] from a drag-drop operation
+     */
+    fun onTabsMove(
```

Comment:
nit: this would be nice to have on the `BrowserTrayInteractor`. The TabsTrayInteractor is generally for "global" tray interactions that apply that would also apply to Synced Tabs as well.

Reply:
It might not be worth moving it at this point though - just a note.

Reply:
The issue with that is the `AbstractBrowserTrayList` doesn't seem to have a `BrowserTrayInteractor`. I could add one but that didn't seem reasonable.

Also, shouldn't a tab reordering be represented in synced tabs? I would expect the tabs list to remain in the same order between devices, and if rearranged the movement would be copied to the other on the next sync.